Okta.Core/OktaHttpClient.cs

```diff
@@ -175,6 +176,24 @@ public override Task<HttpResponseMessage> ExecuteAsync(HttpRequestType requestTy
                     {
                         httpClient.DefaultRequestHeaders.Remove("Authorization");
                     }
+                    if (bAddXForwardedHeader)
+                    {
+                        string ipAddress = null;
+                        if (!string.IsNullOrEmpty(content))
+                        {
+                            content = content.Trim();
+                            if (content.StartsWith("{") && content.EndsWith("}"))
+                            {
+                                var obj = JObject.Parse(content);
+                                ipAddress = (string)obj["context"]["ipAddress"];

```

Comment:
I think that would be a bit cleaner than checking for brackets. Something along the lines of:
```csharp
try {
  var parsedContent = JObject.Parse(content);
  ipAddress = (string)parsedContent["context"]["ipAddress"];
  ...
} catch { /* no ipAddress given */}
```